src/Cardano/WalletLayer/Wallet.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE TypeOperators #-}
+
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides internal implementation of
+-- | wallet api
+
+module Cardano.WalletLayer.Wallet
+    (
+      CreateWallet
+    , WalletId
+    ) where
+
+import Prelude
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.Wallet.AddressDerivation
+    ( Passphrase )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap )
+import Cardano.Wallet.Mnemonic
+    ( Mnemonic )
+import Data.Text
+    ( Text )
+import Data.Time.Units
+    ( Microsecond )
+import GHC.Generics
+    ( Generic )
+
+-- | Errors
+data CreateWalletError
+
+data GetWalletError
+
+-- | Types
+data CreateWallet = CreateWallet NewWallet | ImportWallet
+
+data NewWallet = NewWallet
+    {
+      _mnemonicSentence
+        :: !(Mnemonic 15)
+    , _mnemonicSentencePassphrase
+        :: !(Maybe (Passphrase "generation"))
```

Comment:
The generation passphrase is represented as a short mnemonic (cf the API swagger spec). So we probably want to keep it the same at this stage and make the conversion later.

Reply:
now using `Mnemonic 9`